app/src/main/java/com/example/zerobin/ui/mypage/MyPageViewModel.kt

```diff
@@ -1,35 +1,138 @@
 package com.example.zerobin.ui.mypage
 
 
+import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.zerobin.data.repository.User.UserRepository
+import com.example.zerobin.data.repository.mypage.MypageReviewRepository
+import com.example.zerobin.data.repository.mypage.MypageShopRepository
+import com.example.zerobin.data.repository.mypage.MypageStampRepository
+import com.example.zerobin.domain.DataResult
 import com.example.zerobin.domain.entity.Review
 import com.example.zerobin.domain.entity.Shop
 import com.example.zerobin.domain.entity.User
+import com.example.zerobin.ui.home.HomeViewModel
+import com.example.zerobin.ui.review.ReviewViewModel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
 
 class MyPageViewModel : ViewModel() {
 
 
     private val _myUser = MutableLiveData<User>()
+    private val _myUserReview = MutableLiveData<List<Review>>()
+    private val _myUserShop= MutableLiveData<List<Shop>>()
+    private val _myUserStamp= MutableLiveData<List<Review>>()
 
     val myUser: LiveData<User> = _myUser
+    val myUserReview: LiveData<List<Review>> = _myUserReview
+    val myUserShop: LiveData<List<Shop>> = _myUserShop
+    val myUserStamp: LiveData<List<Review>> = _myUserStamp
+
+
+    private val mypageReviewRepository = MypageReviewRepository()
+    private val mypageShopRepository= MypageShopRepository()
+    private val userRepository = UserRepository()
+    private val mypageStampRepository = MypageStampRepository()
 
     fun requestMyPage() {
 
-        val tempReview1 = Review("땡상점", "ㅋㅋㅋ", listOf("sss","Ashhhhdf"), listOf("22229988","Asdf"), "asdfasfasfasdfsdf", nickName = "asdfad",reviewIndex = 0,stamp=true)
-        val tempReview2 = Review("나영상점", "ㅋㅋㅋ", listOf("sss","Ashhhhdf"), listOf("22229988","Asdf"), "asdfasfasfasdfsdf", nickName = "asdfad",reviewIndex = 0,stamp=true)
-        val listReview: ArrayList<Review> = arrayListOf(tempReview1, tempReview2)
-//        val tempShop1 =
-//            Shop("알맹상점", "#친환경  #리필스테이션  #비건화장품", true, "서울특별시 마포구 월드컵로 49 2층", "https://image")
-//        val tempShop2 =
-//            Shop("알맹상점", "#친환경  #리필스테이션  #비건화장품", true, "서울특별시 마포구 월드컵로 49 2층", "https://image")
-//        val listShop: ArrayList<Shop> = arrayListOf(tempShop1, tempShop2)
-
-        _myUser.value = User(
-            "ny@gamil", "ny", 0, listReview.size, listReview,
-            emptyList()
-        )
+        viewModelScope.launch {
+            val userResponse=userRepository.getMypageUser()
+            userResponse.collect { handleResultUser(it)}
+
+
+        }
+
+
+    }
+    fun requestMyPageReview(){
+        viewModelScope.launch {
+            val response = mypageReviewRepository.getMypageReview()
+            Log.d(HomeViewModel.TAG, response.toString())
+            response.collect { handleResult(it)}
```

Comment:
마이페이지 리뷰 응답에 대한건데 handleResult로  되어있어서 헷갈릴수 있으니 아래처럼 수정하는게 좋을거같아요
```suggestion
            response.collect { handleResultReview(it)}
```